### PR TITLE
Ensure tests can run in isolation

### PR DESCRIPTION
**Pull request by Vinnl**
The primary change here is to change `resetMocks` and `resetModules` to `true`, to ensure each unit test can run independently. To do so, I did have to update the tests from https://github.com/mozilla/blurts-server/pull/3304 to define the mocks each specific test depended on inside that test, rather than in a `beforeAll` call that set them up in the precise order that the tests had to run in as well.

I wasn't entirely sure from the test descriptions what behaviour they were verifying, so I updated the descriptions to specify what I _though_ they did. Let me know if they're wrong.

### CODE REVIEW

File: jest.config.cjs [hunk 1]
```diff
@@ -225,12 +225,12 @@ const customJestConfig = {
       statements: 68,
     },
     "src/cloud-functions/index.js": {
-      branches: 63,
+      branches: 60,
```
Comment (Vinnl) [inline jest.config.cjs:228]: The annoying thing here is that I don't know why this coverage went down, given that I did not actually touch that file :/ If anyone has any thoughts on that, let me know.

File: src/utils/subscriberBreaches.ts [hunk 2]
```diff
@@ -74,7 +74,13 @@ export async function getSubBreaches(
     const breachResolution = subscriber.breach_resolution
```
Comment (mansaj) [inline src/utils/subscriberBreaches.ts:74]: Yea I think this is off.. it should just be something like:
`subscriber.breach_resolution?.[email.email] ?? {}`

Comment (mansaj) [reply src/utils/subscriberBreaches.ts:74]: Good catch!

**Review by rhelmer — APPROVED** [review]

File: jest.config.cjs (hunk 1 above)
Comment (mansaj) [reply jest.config.cjs:228]: We can always open up the `index.html` in coverage folder to see exactly which branches are no longer being tested. But it's probably due to changes to underlying functions (which I don't see any connections between this file and `index.js`.. weird) that file depends on

**Review by mansaj — APPROVED** [review]

File: jest.config.cjs (hunk 1 above)
Comment (Vinnl) [reply jest.config.cjs:228]: Unfortunately I see no changes when eyeballing it, other than the actual number:

Screenshot 2023-08-18 at 16-23-12 Code coverage report for src_cloud-functions_index js
Screenshot 2023-08-18 at 16-23-23 Code coverage report for src_cloud-functions_index js

File: src/utils/subscriberBreaches.ts (hunk 2 above)
Comment (Vinnl) [reply src/utils/subscriberBreaches.ts]: Cool, added in 143426815. Catching stuff like this is what the coverage report is useful for!

**Pull request merged**